app/services/create_github_repo_service.rb

```diff
@@ -59,6 +68,35 @@ def create_github_repository!
     raise Result::Error.new errors(:repository_creation_failed), error.message
   end
 
+  # Public: Clone the GitHub template repository for the Assignment.
+  #
+  # Returns an Integer ID or raises a Result::Error
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  def create_github_repository_from_template!
+    stats_sender.report_with_exercise_prefix(:import_with_templates_started)
+    client = assignment.creator.github_client
+    options = {
+      name: exercise.repo_name,
+      owner: exercise.organization_login,
+      private: assignment.private?,
+      description: "#{exercise.repo_name} created by GitHub Classroom",
+      accept: TEMPLATE_REPOS_API_PREVIEW,
+      include_all_branches: true
+    }
+
+    github_repository = client.post(
```

Comment:
Are we able to move this GitHub logic to `github_repository`?

Reply:
@EricPickup and I had a chat on this, we are waiting for the Octokit upgrade and we are then going to update this in a separate PR, is that okay?

Reply:
Yup, I saw that PR after I left this comment. No problem :)

Reply:
Also created an issue for this, tagging here for reference: #2075